### PR TITLE
🐛 Source Sendgrid: add start_time config and correct primary_key

### PR DESCRIPTION
**Pull request by marcosmarxm**
## What
*Describe what the change is solving*
*It helps to add screenshots if it affects the frontend.*

## How
Add config `start_time` in spec.json and correct primary_key value to `email`

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist
Expand the checklist which is relevant for this PR. 

<details><summary> <strong> Connector checklist </strong> </summary>
<p>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Secrets are annotated with `airbyte_secret` in the connector's spec
- [ ] Credentials added to Github CI if needed and not already present. [instructions for injecting secrets into CI](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#using-credentials-in-ci). 
- [ ] Unit & integration tests added as appropriate (and are passing)
    * Community members: please provide proof of this succeeding locally e.g: screenshot or copy-paste acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] `/test connector=connectors/<name>` command as documented [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-an-existing-connector) is passing. 
    * Community members can skip this, Airbyters will run this for you. 
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] `README.md`
    - [ ] `docs/SUMMARY.md` if it's a new connector
    - [ ] Created or updated reference docs in `docs/integrations/<source or destination>/<name>`.
    - [ ] Changelog in the appropriate page in `docs/integrations/...`. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md` contains a reference to the new connector
    - [ ] Build status added to [build page](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/builds.md)
- [ ] Build is successful
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] No major blockers
- [ ] PR merged into master branch
- [ ] Follow up tickets have been created
- [ ] Associated tickets have been closed & stakeholders notified
</p>
</details>

<details><summary> <strong> Connector Generator checklist </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
@sherifnada I'm going to merge before review because one user is trying this source right now.
image

File: airbyte-integrations/connectors/source-sendgrid/sample_files/no_spam_reports_configured_catalog.json [hunk 1]
```diff
@@ -411,32 +382,6 @@
       "sync_mode": "incremental",
       "cursor_field": ["created"],
       "destination_sync_mode": "append"
-    },
```
Comment (keu) [inline airbyte-integrations/connectors/source-sendgrid/sample_files/no_spam_reports_configured_catalog.json:414]: why these two were removed?

File: airbyte-integrations/connectors/source-sendgrid/source_sendgrid/spec.json [hunk 2]
```diff
@@ -10,6 +10,10 @@
       "apikey": {
         "type": "string",
         "description": "API Key, use <a href=\"https://app.sendgrid.com/settings/api_keys/\">admin</a> to generate this key."
+      },
+      "start_time": {
+        "type": "integer",
+        "description": "Start time in timestamp integer format, example: 1558359837"
```
Comment (keu) [inline airbyte-integrations/connectors/source-sendgrid/source_sendgrid/spec.json:16]:
```suggestion
        "description": "Start time in timestamp integer format. Any data before this date will not be replicated.",
        "examples": [1558359837]
```

File: airbyte-integrations/connectors/source-sendgrid/sample_files/no_spam_reports_configured_catalog.json (hunk 1 above)
Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-sendgrid/sample_files/no_spam_reports_configured_catalog.json:414]: they are returning zero records in tests and break it.

File: airbyte-integrations/connectors/source-sendgrid/source_sendgrid/streams.py [hunk 3]
```diff
@@ -32,7 +32,7 @@
 
 class SendgridStream(HttpStream, ABC):
     url_base = "https://api.sendgrid.com/v3/"
-    primary_key = "id"
+    primary_key = "email"
```
Comment (keu) [inline airbyte-integrations/connectors/source-sendgrid/source_sendgrid/streams.py:35]: I don't think this is correct, some streams don't have `email` attribute, but have `id`, I think you need to check each stream and override primary_key = "email" where necessary, WDYT?

Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-sendgrid/source_sendgrid/streams.py:35]: all incremental streams at the moment use `email` other only have `full_refresh` enabled.

**Review by keu — CHANGES_REQUESTED** [review]
see my comments

File: airbyte-integrations/connectors/source-sendgrid/source_sendgrid/streams.py (hunk 3 above)
Comment (keu) [reply airbyte-integrations/connectors/source-sendgrid/source_sendgrid/streams.py:35]: in such case maybe it is better to leave primary_key=id, and override Incremental base class?

File: airbyte-integrations/connectors/source-sendgrid/sample_files/no_spam_reports_configured_catalog.json (hunk 1 above)
Comment (keu) [reply airbyte-integrations/connectors/source-sendgrid/sample_files/no_spam_reports_configured_catalog.json:414]: I see, then I think the name of the file slightly off

**Review by keu — APPROVED** [review]

**Pull request merged**